### PR TITLE
feat(extras): improve ruby extra by letting user chose

### PR DESCRIPTION
**Pull request by kevinrobayna**
## What is this PR for?

Shopify started working on its own LSP (https://github.com/Shopify/ruby-lsp) and it performs way better than Solargraph which has a lot of limitations. This paired with sorbet gives better IntelliSense when navigating the code.

This PR follows the same approach as Python and lets the user configure through vim.g options the lsp and formatter for ruby, without overriding any configuration.

## Does this PR fix an existing issue?

One caveat though is that RubyLsp does not work very well with NeoVim < 0.10 https://github.com/Shopify/ruby-lsp/blob/main/EDITORS.md#neovim

## Checklist

- [x] I've read the [CONTRIBUTING](https://github.com/LazyVim/LazyVim/blob/main/CONTRIBUTING.md) guidelines.

### CODE REVIEW

**Comment by kevinrobayna** [comment]
Removed sorbet because if your ruby project does not have it setup you'd get warnings so it's better if you add it yourself

**Comment by folke** [comment]
Can you change it similar to how it's done for python?
So add a doc section to the top and a `vim.g` var to chose.
I'm fine with that flag defaulting to `rubylsp` on NEovim >= 0.10

**Comment by kevinrobayna** [comment]
> Can you change it similar to how it's done for python? So add a doc section to the top and a `vim.g` var to chose. I'm fine with that flag defaulting to `rubylsp` on NEovim >= 0.10

Thank you for the idea @folke this is soo much better i took the opportunity to also make the formatter use this

File: lua/lazyvim/plugins/extras/lang/ruby.lua [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+  -- LSP Server to use for Ruby.
+  -- Set to "solargraph" to use solargraph instead of ruby_lsp.
+  vim.g.lazyvim_ruby_lsp = "ruby_lsp"
+  vim.g.lazyvim_ruby_formatter = "rubocop"
+end
+
+local lsp = vim.g.lazyvim_ruby_lsp or "ruby_lsp"
+if vim.fn.has("nvim-0.10") == 0 then
+  -- ruby_lsp does not work well with Neovim < 0.10
+  lsp = vim.g.lazyvim_ruby_lsp or "solarpath"
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/lang/ruby.lua:11]: Shouldn't this be `solargraph`?

Comment (kevinrobayna) [reply lua/lazyvim/plugins/extras/lang/ruby.lua]: yes! ty for the feedback1

**Comment by folke** [comment]
ty!

**Comment by ahmedsayedabdelsalam** [comment]
how to make rails model validation auto completion work?

**Pull request merged**